WordPress/Sniffs/WP/CapabilitiesSniff.php

```diff
@@ -0,0 +1,470 @@
+	 * The functions are defined in `wp-admin/includes/plugin.php` and
+	 * `/wp-includes/capabilities.php`.
+	 * The list is sorted alphabetically.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var array<string, array> The key is the name of a function we're targetting,
+	 *                           the value is an array containing the 1-based parameter position
+	 *                           of the "capability" parameter within the function, as well as
+	 *                           the name of the parameter as declared in the function.
+	 *                           If the parameter name has been renamed since the release of PHP 8.0,
+	 *                           the parameter can be set as an array.
+	 */
+	protected $target_functions = array(
+		'add_comments_page'         => array(
+			'position' => 3,
+			'name'     => 'capability',
+		),
+		'add_dashboard_page'        => array(
+			'position' => 3,
+			'name'     => 'capability',
+		),
+		'add_links_page'            => array(
+			'position' => 3,
+			'name'     => 'capability',
+		),
+		'add_management_page'       => array(
+			'position' => 3,
+			'name'     => 'capability',
+		),
+		'add_media_page'            => array(
+			'position' => 3,
+			'name'     => 'capability',
+		),
+		'add_menu_page'             => array(
+			'position' => 3,
+			'name'     => 'capability',
+		),
+		'add_object_page'           => array( // Deprecated since WP 4.5.0.
+			'position' => 3,
+			'name'     => 'capability',
+		),
+		'add_options_page'          => array(
+			'position' => 3,
+			'name'     => 'capability',
+		),
+		'add_pages_page'            => array(
+			'position' => 3,
+			'name'     => 'capability',
+		),
+		'add_plugins_page'          => array(
+			'position' => 3,
+			'name'     => 'capability',
+		),
+		'add_posts_page'            => array(
+			'position' => 3,
+			'name'     => 'capability',
+		),
+		'add_submenu_page'          => array(
+			'position' => 4,
+			'name'     => 'capability',
+		),
+		'add_theme_page'            => array(
+			'position' => 3,
+			'name'     => 'capability',
+		),
+		'add_users_page'            => array(
+			'position' => 3,
+			'name'     => 'capability',
+		),
+		'add_utility_page'          => array( // Deprecated since WP 4.5.0.
+			'position' => 3,
+			'name'     => 'capability',
+		),
+		'author_can'                => array(
+			'position' => 2,
+			'name'     => 'capability',
+		),
+		'current_user_can'          => array(
+			'position' => 1,
+			'name'     => 'capability',
+		),
+		'current_user_can_for_blog' => array(
+			'position' => 2,
+			'name'     => 'capability',
+		),
+		'map_meta_cap'              => array(
+			'position' => 1,
+			'name'     => 'cap',
+		),
+		'user_can'                  => array(
+			'position' => 2,
+			'name'     => 'capability',
+		),
+	);
+
+	/**
+	 * List of core roles which should not to be used directly.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var array<string, bool> Role available in WP Core.
+	 */
+	protected $core_roles = array(
```

Comment:
Any particular reason why this property is `protected` ? 

```suggestion
	private $core_roles = array(
```